battlepass/src/lib.rs

```diff
@@ -0,0 +1,304 @@
+		pub fn close_battlepass(
+			origin: OriginFor<T>,
+			battlepass_id: T::Hash,
+		) -> DispatchResult {
+			let sender = ensure_signed(origin)?;
+			// check if Battlepass exists
+			let battlepass = Self::get_battlepass(battlepass_id).ok_or(Error::<T>::BattlepassUnknown)?;
+			// check if Battlepass is active
+			ensure!(Self::is_battlepass_active(battlepass_id), Error::<T>::BattlepassInactive);
+			// check if Org is active
+			ensure!(T::Control::is_org_active(&battlepass.org_id), Error::<T>::OrgUnknownOrInactive);
+			// check if origin is an Org Prime
+			let prime = T::Control::org_prime_account(&battlepass.org_id).ok_or(Error::<T>::OrgPrimeUnknown)?;
+			ensure!(sender == prime, Error::<T>::AuthorizationError);  // TODO: Add Root temporarily
+
+			Self::do_close_battlepass(battlepass.org_id, battlepass_id)?;
+
+			Self::deposit_event(Event::BattlepassClosed { closed_by: sender, org_id: battlepass.org_id, battlepass_id });
+
+			Ok(())
+		}
+
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	fn create_collection(owner: T::AccountId) -> Result<u32, DispatchError> {
+		let metadata = BoundedVec::truncate_from(b"meta".to_vec());		// TODO: what should be here?
+		let symbol = BoundedVec::truncate_from(b"symbol".to_vec());		// TODO: what should be here?
+		let collection_id = T::Rmrk::collection_create(owner, metadata, None, symbol)?;
+
+		Ok(collection_id)
+	}
+
+	fn is_battlepass_active(battlepass_id: T::Hash) -> bool {
+		if let Some(bp_state) = Self::get_battlepass_state(battlepass_id) {
+			return bp_state == BattlepassState::ACTIVE;
+		} 
+		return false;
+	}
+
+	fn get_battlepass_info(org_id: T::Hash) -> (u32, Option<T::Hash>) {
+		if let Some(bp_info) = BattlepassInfoByOrg::<T>::get(&org_id) {
+			return (bp_info.count, bp_info.active);
+		} else {
+			return (0, None);
+		}
+	}
+	
+	fn do_create_battlepass(creator: T::AccountId, org_id: T::Hash, name: String<T>, cid: String<T>, collection_id: u32, price: u16, new_season:u32) -> Result<T::Hash, DispatchError> {
+		let battlepass: Battlepass<T::Hash, T::AccountId, String<T>> = types::Battlepass {
+			creator, 
+			org_id,
+			name,
+			cid,
+			season: new_season,
+			collection_id,
+			price
+		};
+		let battlepass_id = <T as frame_system::Config>::Hashing::hash_of(&battlepass);
+
+		Battlepasses::<T>::insert(&battlepass_id, battlepass);
+		BattlepassStates::<T>::insert(&battlepass_id, types::BattlepassState::ACTIVE); 					// TODO: change to Draft
+		BattlepassInfoByOrg::<T>::insert(org_id, BattlepassInfo{count: new_season, active: Some(battlepass_id)});
+
+		Ok(battlepass_id)
+	}
+
+	fn do_claim_battlepass(claimer: T::AccountId, battlepass_id: T::Hash, nft_id: u32, collection_id: u32) -> DispatchResult {
+
+		// Create Battlepass NFT
+		// let metadata = battlepass_id.to_string().into_bytes();
+		let metadata = b"meta".to_vec();
+		let (_, nft_id) = T::Rmrk::nft_mint(
+			claimer.clone(),									// sender
+			claimer.clone(),										// owner
+			nft_id,														// nft_id
```

Comment:
this should be unique hash for the actual nft hash

Reply:
This is an ID (u32) of NFT in the Battlepass' Collection.

Reply:
during minting the nft gets its unique hash, based on the docs this should be it, or where is the hash created?

Reply:
Which docs you are referring to? As I can see, RMRK doesn't create a hash. The key for NFT is collection_id + nft_id.